pkg/agent/cniserver/server_linux_test.go

```diff
@@ -0,0 +1,743 @@
+		)
+	})
+
+	t.Run("Interface check failure veth", func(t *testing.T) {
+		cniConfig := baseCNIConfig()
+		cniConfig.Ifname = ifname
+		cniConfig.Netns = "invalid_netns"
+		sriovVFDeviceID := ""
+		cniServer.podConfigurator, _ = newPodConfigurator(nil, nil, nil, nil, nil, "", false, channel.NewSubscribableChannel("PodUpdate", 100), nil, false)
+		response := cniServer.validatePrevResult(cniConfig.CniCmdArgs, prevResult, sriovVFDeviceID)
+		checkErrorResponse(t, response, cnipb.ErrorCode_CHECK_INTERFACE_FAILURE, "")
+	})
+
+	t.Run("Interface check failure SR-IOV VF", func(t *testing.T) {
+		cniConfig := baseCNIConfig()
+		cniConfig.Ifname = ifname
+		cniConfig.Netns = "invalid_netns"
+		sriovVFDeviceID := "0000:03:00.6"
+		prevResult.Interfaces = []*current.Interface{hostIface, containerIface}
+		cniServer.podConfigurator, _ = newPodConfigurator(nil, nil, nil, nil, nil, "", true, channel.NewSubscribableChannel("PodUpdate", 100), nil, false)
+		response := cniServer.validatePrevResult(cniConfig.CniCmdArgs, prevResult, sriovVFDeviceID)
+		checkErrorResponse(t, response, cnipb.ErrorCode_CHECK_INTERFACE_FAILURE, "")
+	})
+}
+
+func TestRemoveInterface(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	mockOVSBridgeClient = ovsconfigtest.NewMockOVSBridgeClient(controller)
+	mockOFClient = openflowtest.NewMockClient(controller)
+	ifaceStore = interfacestore.NewInterfaceStore()
+	routeMock = routetest.NewMockInterface(controller)
+	gwMAC, _ := net.ParseMAC("00:00:11:11:11:11")
+	podConfigurator, err := newPodConfigurator(mockOVSBridgeClient, mockOFClient, routeMock, ifaceStore, gwMAC, "system", false, channel.NewSubscribableChannel("PodUpdate", 100), nil, false)
+	require.Nil(t, err, "No error expected in podConfigurator constructor")
+
+	containerMAC, _ := net.ParseMAC("aa:bb:cc:dd:ee:ff")
+	containerIP := net.ParseIP("1.1.1.1")
+
+	var cniConfig *CNIConfig
+	var containerID string
+	var podName string
+	var hostIfaceName string
+	var fakePortUUID string
+	var containerConfig *interfacestore.InterfaceConfig
+
+	setup := func(name string) {
+		containerID = uuid.New().String()
+		podName = name
+		hostIfaceName = util.GenerateContainerInterfaceName(podName, testPodNamespace, containerID)
+		fakePortUUID = uuid.New().String()
+
+		netcfg := generateNetworkConfiguration("", supportedCNIVersion, "", testIpamType)
+		cniConfig = &CNIConfig{NetworkConfig: netcfg, CniCmdArgs: &cnipb.CniCmdArgs{}}
+		cniConfig.Ifname = "eth0"
+		cniConfig.ContainerId = containerID
+		cniConfig.Netns = ""
+
+		containerConfig = interfacestore.NewContainerInterface(
+			hostIfaceName,
+			containerID,
+			podName,
+			testPodNamespace,
+			containerMAC,
+			[]net.IP{containerIP},
+			0)
+		containerConfig.OVSPortConfig = &interfacestore.OVSPortConfig{PortUUID: fakePortUUID, OFPort: 0}
+	}
+
+	t.Run("Successful removal", func(t *testing.T) {
+		setup("test1")
+		ifaceStore.AddInterface(containerConfig)
```

Comment:
setup is too obsecure to understand what it does to set up, while it seems it just create a containerConfig, then why don't just having a function `newContainerConfig()` and returning required variables to be more readable and explicit?